src/Controls/src/Core/MultiBinding.cs

```diff
@@ -153,26 +119,19 @@ internal override void Apply(object context, BindableObject targetObject, Bindab
 						var bp = _bpProxies[i] = BindableProperty.Create($"mb-proxy{i}", typeof(object), typeof(MultiBinding), null, bindingMode, propertyChanged: OnBindingChanged);
 						_proxyObject.SetBinding(bp, binding);
 					}
-					_applying = false;
 				}
 			}
-			_proxyObject.BindingContext = context;
-
-			if (this.GetRealizedMode(_targetProperty) == BindingMode.OneWayToSource)
-				return;
-
-			var value = GetSourceValue(GetValueArray(), _targetProperty.ReturnType);
-			if (value != Binding.DoNothing)
+			else
 			{
-				_applying = true;
-				if (!BindingExpression.TryConvert(ref value, _targetProperty, _targetProperty.ReturnType, true))
-				{
-					BindingDiagnostics.SendBindingFailure(this, context, _targetObject, _targetProperty, "MultiBinding", BindingExpression.CannotConvertTypeErrorMessage, value, _targetProperty.ReturnType);
-					return;
-				}
-				_targetObject.SetValueCore(_targetProperty, value, SetValueFlags.ClearDynamicResource, BindableObject.SetValuePrivateFlags.Default | BindableObject.SetValuePrivateFlags.Converted);
-				_applying = false;
+				_updates = false;
+				_proxyObject.BindingContext = context;
 			}
+			_applying = false;
+
+			if (this.GetRealizedMode(_targetProperty) == BindingMode.OneWayToSource || !_updates )
+				return;
```

Comment:
If `_updates` had a clearer name, maybe others would understand why we are returning early here.

Reply:
See above.